src/diagnostics.rs

```diff
@@ -157,6 +157,19 @@ pub fn report_error<'tcx, 'mir>(
         }
     };
 
+    let stacktrace = ecx
+        .generate_stacktrace()
+        .into_iter()
+        .take_while(|e| {
+            let def_id = e.instance.def_id();
+            let path = ty::tls::with_opt(|opt_tcx| {
+                opt_tcx.map(|tcx| tcx.def_path(def_id).to_string_no_crate_verbose())
+            })
+            .unwrap_or_default();
+            !path.contains("__rust_begin_short_backtrace")
```

Comment:
What about `__rust_end_short_backtrace`? This doesn't look equivalent to the logic in std.

Reply:
`__rust_begin_short_backtrace` and `__rust_end_short_backtrace` are special symbols in `std` that exist _only_ so that they can be relied upon to appear in the backtrace and serve to bound the interesting region of the backtrace.

`__rust_end_short_backtrace` is called by the default panic handler. I think I was wrong to use these symbols; we're missing the `end` one because we're not in a panic, and in any case we don't have to rely on special symbols. We can ask if the `DefId` is local to skip over the Rust runtime completely.

Reply:
Things could be not local but still be in another crate, not part of the Rust runtime.

So I think relying on these special symbols makes a lot of sense.

Reply:
Oh I see what you mean, you want to skip only bottom non-local frames below the first local frame.

How does that look like, for a binary and for a test? Could you update the PR description?

Reply:
:facepalm: good thing to ask for. I didn't notice the test runtime passes through a local symbol on its way to the actual test.